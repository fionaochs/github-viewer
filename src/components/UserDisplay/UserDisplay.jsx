import React from 'react';
import PropTypes from 'prop-types';

const UserDisplay = ({ userName, followers, following, url }) => (
  <ul>
    <li key={userName}>
      <p>{userName}</p>
      <p>{followers}</p>
      <p>{following}</p>
      <a href={url}>Link to github</a>
    </li>
  </ul>
);

UserDisplay.propTypes = {
  userName: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default UserDisplay;
