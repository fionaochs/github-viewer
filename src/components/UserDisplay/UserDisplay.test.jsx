import React from 'react';
import { shallow } from 'enzyme';
import UserDisplay from './UserDisplay';

describe('UserDisplay', () => {
  it('matches a user display snapshot', () => {
    const wrapper = shallow(<UserDisplay userName='me'
      followers='50'
      following='50'
      link='https://rickandmortyapi.com/api/UserDisplay/avatar/1.jpeg' />);
    expect(wrapper).toMatchSnapshot();
  });
});
