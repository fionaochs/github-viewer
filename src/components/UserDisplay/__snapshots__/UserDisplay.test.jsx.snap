// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay matches a user display snapshot 1`] = `
<ul>
  <li
    key="me"
  >
    <p>
      me
    </p>
    <p>
      50
    </p>
    <p>
      50
    </p>
    <a>
      Link to github
    </a>
  </li>
</ul>
`;
